docs/content/1.documentation/5.advanced/2.faq.md

```diff
@@ -245,43 +245,14 @@ Next, you need to configure your img tag to include the `crossorigin` attribute:
 ℹ Read more about it [here](https://github.com/Baroshem/nuxt-security/issues/138#issuecomment-1497883915).
 ::
 
-## Using nonce with CSP for Nuxt Image
+## Nuxt Image
 
-Having securely configured images is crucial for modern web applications. Check out how to do it below:
-
-```ts
-// nuxt.config.ts
-
-security: {
-  nonce: true,
-  headers: {
-    contentSecurityPolicy: {
-      'img-src': ["'self'", 'data:', 'https:'],
-      'script-src': [
-        "'self'", // backwards compatibility for older browsers that don't support strict-dynamic
-        "'nonce-{{nonce}}'",
-        "'strict-dynamic'"
-      ],
-      'script-src-attr': ["'self'"]
-    }
-  }
-}
-```
-
-And then configure `NuxtImg` like following:
-
-```vue
-<template>
-  <NuxtImg src="https://localhost:8000/api/image/xyz" :nonce="nonce" />
-</template>
-
-<script lang="ts" setup>
-const nonce = useNonce()
-</script>
-```
+When using `<NuxtImg>` or `<NuxtPicture>`, an inline script will be used for error handling during SSR.
+This will lead to CSP issues if `unsafe-inline` is not allowed and the image fails to load.
```

Comment:
Hey @P4sca1 
Isn’t it a bit severe ?
You could probably use ‘unsafe-hashes’ here, and the inline code is always the same so you could pre-hash it.
I do agree this is not ideal though. @harlan-zw was able to replace all inline event handlers with addEventListener in @nuxt/scripts so maybe the team at NuxtImg can use the same approach?

Reply:
The inline code will indeed be always the same, so using `unsafe-hashes` could work. Maybe we could add it by default in non strict mode or behind a feature flag in strict mode to support `<NuxtImg>` and `<NuxtPicture>`.

I calculated the hash that would be needed:
```sh
echo -n "this.setAttribute('data-error', 1)" | openssl sha256 -binary | openssl base64
bwK6T5wZVTANitXbrTsel7kl/PyCjCd/Dq5Qoz3imjM=
```

Using `addEventListener` in this case is not trivial, because the event listener would be attached in `onMounted()`, which is too late for some kind of errors. So some errors, e.g. when the url is invalid, could be missed.

Reply:
Hey @P4sca1 
What is the issue when CSP denies execution of the error handler?

Reply:
The `data-error` property does not get set on the image tag and the error event is never emitted.
From a user's perspective the page works fine, it just shows an unloaded image.